controllers/zync_controller.go

```diff
@@ -0,0 +1,165 @@
+// +kubebuilder:rbac:groups="core",namespace=placeholder,resources=secrets,verbs=get;list;watch
+// +kubebuilder:rbac:groups="apps",namespace=placeholder,resources=deployments,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="monitoring.coreos.com",namespace=placeholder,resources=podmonitors,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="autoscaling",namespace=placeholder,resources=horizontalpodautoscalers,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="policy",namespace=placeholder,resources=poddisruptionbudgets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="integreatly.org",namespace=placeholder,resources=grafanadashboards,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups="secrets-manager.tuenti.io",namespace=placeholder,resources=secretdefinitions,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *ZyncReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+
+	instance := &saasv1alpha1.Zync{}
+	key := types.NamespacedName{Name: req.Name, Namespace: req.Namespace}
+	result, err := r.GetInstance(ctx, key, instance, saasv1alpha1.Finalizer, log)
+	if result != nil || err != nil {
+		return *result, err
+	}
+
+	// Apply defaults for reconcile but do not store them in the API
+	instance.Default()
+	json, _ := json.Marshal(instance.Spec)
+	log.V(1).Info("Apply defaults before resolving templates", "JSON", string(json))
+
+	gen := zync.NewGenerator(
+		instance.GetName(),
+		instance.GetNamespace(),
+		instance.Spec,
+	)
+
+	// Calculate rollout triggers
+	triggers, err := r.TriggersFromSecretDefs(ctx,
+		gen.ZyncSecretDefinition(),
+	)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	err = r.ReconcileOwnedResources(ctx, instance, basereconciler.ControlledResources{
+		Deployments: []basereconciler.Deployment{
+			{
+				Template:        gen.API.Deployment(),
+				HasHPA:          !instance.Spec.API.HPA.IsDeactivated(),
+				RolloutTriggers: triggers,
+			},
+			{
+				Template:        gen.Que.Deployment(),
+				HasHPA:          !instance.Spec.Que.HPA.IsDeactivated(),
+				RolloutTriggers: triggers,
+			},
+		},
+		SecretDefinitions: []basereconciler.SecretDefinition{
+			{
+				Template: gen.ZyncSecretDefinition(),
+				Enabled:  true,
+			},
+		},
+		Services: []basereconciler.Service{
+			{
+				Template: gen.API.Service(),
+				Enabled:  true,
+			}},
+		PodDisruptionBudgets: []basereconciler.PodDisruptionBudget{
+			{
+				Template: gen.API.PDB(), // Calculate rollout triggers
+				Enabled:  !instance.Spec.API.PDB.IsDeactivated(),
+			},
+			{
+				Template: gen.Que.PDB(), // Calculate rollout triggers
```

Comment:
```suggestion
				Template: gen.Que.PDB(),
```

Reply:
fixed in https://github.com/3scale/saas-operator/pull/74/commits/5c48490b93a3f1f4a1e57fe254ef0ff4cecce725